app/models/user.rb

```diff
@@ -1,14 +1,30 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
 
-  after_create do 
-    if User.count == 1
-      self.admin = true
-      self.save
+  before_create :set_admin, :create_slug
+  validates :email, uniqueness: true
+
+  enum role: [:admin, :author, :editor]
+  after_initialize :set_default_role, if: :new_record?
+
+  def set_default_role
+    self.role ||= :author
+  end
+
+  def create_slug
+    if full_name
+      full_name_split = self.full_name.split(" ");
+      self.slug = full_name_split[0].to_s + "-" + full_name_split[1]&.first.to_s;
```

Comment:
make it shorter. Maybe make full_name required or set its default value to be an empty string - but then you will need safe operators